orange/orange.lua

```diff
@@ -144,13 +152,87 @@ function Orange.access()
         plugin.handler:access()
     end
 
+    local upstream_url = ngx.var.upstream_url
+    ngx.log(ngx.INFO, "[AFTER ACCESS] ", " upstream_url: " , upstream_url)
+
+    -- here we set the ngx.var.target
+    local target = upstream_url
+    local scheme, hostname
+    local balancer_address
+    if string_find(upstream_url, "://") then
+        scheme, hostname = upstream_url:match("^(.+)://(.+)$")
+    else
+        schema = "http"
+        hostname = upstream_url
+    end
+
+    ngx.log(ngx.INFO, "[scheme] ", scheme, "; [hostname] ", hostname)
+
+    -- only care about upstreams stored in db
+    if utils.hostnameType(hostname) == "name" then
+        local upstreams = orange_db.get_json("balancer.selectors")
```

Comment:
balancer.selectors will be initialized in Orange.init like other plugins. I want the balancer to reuse the selector and rules structure(upstream => selector, hosts => rules), this will make life easier.

you can see the balancer data structure here: https://gist.github.com/zhjwpku/ea6c9036897217abe1f50532b415f93c